src/dss/list.py

```diff
@@ -0,0 +1,54 @@
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.apps_v1 import Deployment
+from tabulate import tabulate
+
+from dss.config import DSS_NAMESPACE, NOTEBOOK_LABEL, DeploymentState
+from dss.logger import setup_logger
+from dss.utils import get_deployment_state, get_service_url, truncate_row
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def list_notebooks(lightkube_client: Client, wide: bool = False) -> None:
+    """
+    List the available notebooks in the DSS namespace.
+
+    Args:
+        lightkube_client (Client): The Kubernetes client.
+        wide (bool, optional): Whether to display the full information without truncation.
+                               Defaults to False.
+    """
+    try:
+        deployments = lightkube_client.list(Deployment, namespace=DSS_NAMESPACE)
+    except ApiError as e:
+        logger.error(f"Failed to list notebooks: {e}")
+        return
+
+    notebook_list = []
+    for deployment in deployments:
+        labels = deployment.metadata.labels
+        if NOTEBOOK_LABEL in labels:
+            name = deployment.metadata.name
+            image = deployment.spec.template.spec.containers[0].image
+            state = get_deployment_state(deployment, lightkube_client)
+
+            # Use state to decide what to display in the URL column
+            if state == DeploymentState.ACTIVE:
+                available_replicas = deployment.status.availableReplicas
+                url = (
+                    get_service_url(name, DSS_NAMESPACE, lightkube_client)
+                    if available_replicas
+                    else f"({DeploymentState.STOPPED})"
+                )
+            else:
+                url = f"({state.value})"
+
+            if not wide:
+                name, image, url = truncate_row(name, image, url)
+
+            notebook_list.append([name, image, url])
```

Comment:
iiuc the `if NOTEBOOK_LEVEL in labels:` part of this is just to filter such that we only inspect `Deployments` that include any label with the key `NOTEBOOK_LABEL`.  If yes, we can simplify this by using the label filtering on `lightkube_client.list()`:
```suggestion
    try:
        deployments = lightkube_client.list(Deployment, namespace=DSS_NAMESPACE, labels={NOTEBOOK_LABEL: lightkube.operators.exists()})
    except ApiError as e:
        logger.error(f"Failed to list notebooks: {e}")
        return

    notebook_list = []
    for deployment in deployments:
        name = deployment.metadata.name
        image = deployment.spec.template.spec.containers[0].image
        state = get_deployment_state(deployment, lightkube_client)

        # Use state to decide what to display in the URL column
        if state == DeploymentState.ACTIVE:
            available_replicas = deployment.status.availableReplicas
            url = (
                get_service_url(name, DSS_NAMESPACE, lightkube_client)
                if available_replicas
                else f"({DeploymentState.STOPPED})"
            )
        else:
            url = f"({state.value})"

        if not wide:
            name, image, url = truncate_row(name, image, url)

        notebook_list.append([name, image, url])
```

This syntax means "fetch any deployment that has any label with key NOTEBOOK_LABEL" ([ref](https://lightkube.readthedocs.io/en/latest/selectors/)).  If you prefer, we can simplify it down to `lightkube_client.list(Deployment, namespace=DSS_NAMESPACE, labels={NOTEBOOK_LABEL: None})` too, but imo the `lightkube.operators.exists()` feels more readable.

Reply:
Sorry, the diff viewer here makes it hard to see the changes.  Might be easier to paste this into your local file.  Its just changing the `.list()` call args, then removing the `if NOTEBOOK_LABEL in labels:` part